### PR TITLE
Rename devworkspace url instances from ideUrl to mainUrl

### PR DESCRIPTION
**Pull request by JPinkney**
Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### What does this PR do?
This PR updates all devworkspace instances from ideUrl to mainUrl.

### What issues does this PR fix or reference?
Part of https://github.com/devfile/devworkspace-operator/issues/399

### CODE REVIEW

File: src/store/__mocks__/devWorkspaceBuilder.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class DevWorkspaceBuilder {
   }
 
   withIdeUrl(ideUrl: string): DevWorkspaceBuilder {
```
Comment (sleshchenko) [inline src/store/__mocks__/devWorkspaceBuilder.ts:58]: Should this method be renamed?

**Review by olexii4 — APPROVED** [review]
LGTM

**Review by akurinnoy — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

**Comment by sleshchenko** [comment]
[dashboard-next-ci-test]

**Comment by sleshchenko** [comment]
[dashboard-next-ci-test]

**Comment by sleshchenko** [comment]
[dashboard-next-ci-test]

**Comment by sleshchenko** [comment]
Tested manually with https://github.com/devfile/devworkspace-operator/pull/423 but it works if I start workspace without factory URL.
Factory URL has an issue but I believe it's not related to this PR, so created a new issue https://github.com/eclipse/che/issues/19800

**Comment by sleshchenko** [comment]
happy path is failing due https://github.com/eclipse-che/che-dashboard/pull/228 which moved dockerfile but tests did not get the corresponding changes. Waiting until happy path test will be adapted.

**Comment by SkorikSergey** [comment]
[dashboard-next-ci-test]

**Comment by SkorikSergey** [comment]
[dashboard-next-ci-test]

**Pull request merged**